cloud/amazon/sts_session_token.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: sts_session_token
+short_description: Obtain a session token from the AWS Security Token Service
+description:
+    - Obtain a session token from the AWS Security Token Service
+version_added: "2.2"
+author: Victor Costan (@pwnall)
+options:
+  duration_seconds:
+    description:
+      - The duration, in seconds, of the session token. The value can range from 900 seconds (15 minutes) to 129600 seconds (36 hours). By default, the value is set to 3600 seconds (1 hour) for an AWS account or 43200 seconds for an IAM account.
+    required: false
+    default: null
+  mfa_serial_number:
+    description:
+      - The identification number of the MFA device that is associated with the user who is making the GetSessionToken call.
+    required: false
+    default: null
+  mfa_token:
+    description:
+      - The value provided by the MFA device, if the trust policy of the user requires MFA.
+    required: false
+    default: null
+notes:
+  - In order to use the session token in a following playbook task you must pass the access_key, access_secret and access_token
```

Comment:
We can make this a little more readable by adding formatting

From http://docs.ansible.com/ansible/developing_modules.html#module-checklist

  * Ensure that you make use of U() for urls, C() for files and options, I() for params, M() for modules.


`  - In order to use the session token in a following playbook task you must pass the I(access_key), I(access_secret) and I(access_token).
`